Sources/tart/Fetcher.swift

```diff
@@ -15,12 +15,28 @@ class Fetcher {
   private static func fetchViaMemory(_ request: URLRequest) async throws -> (AsyncThrowingChannel<Data, Error>, HTTPURLResponse) {
     let dataCh = AsyncThrowingChannel<Data, Error>()
 
-    let (data, response) = try await urlSession.data(for: request)
+    // Use `bytes` instead of `data` here so we can report progress
+    let (asyncBytes, response) = try await urlSession.bytes(for: request)
+
+    let chunkSize = 64 * 1024 * 1024
 
     Task {
-      await dataCh.send(data)
+      let expectedLength = response.expectedContentLength
 
-      dataCh.finish()
+      var chunk = Data()
+      if expectedLength > 0 {
+          chunk.reserveCapacity(min(chunkSize, Int(expectedLength)))
+      } else {
+          chunk.reserveCapacity(chunkSize)
+      }
+
+      for try await byte in asyncBytes {
```

Comment:
We've used this approach in the past, unfortunately, the `URLSession.bytes()` literally processes each byte individually without any buffering, which results in severe performance degradation, see https://github.com/cirruslabs/tart/pull/282.

Reply:
Hehe, yepp, that's what I saw now when testing localhost. It wasn't as visible when testing against the Apple content servers.

Would restoring the delegate work, or is there no way to have progress and speed at the same time?

Reply:
>Would restoring the delegate work, or is there no way to have progress and speed at the same time?

/cc @fkorotkov

Reply:
I think I've found a way to make this work, stay tuned :)